### PR TITLE
Article 002: Camel Case

### PR DESCRIPTION
**Pull request by Teifun2**
#### Length of text ####
:page_with_curl: short
:book:  long

####  Difficulty to read ####
:mouse:  light

####  Difficulty to understand #### 
:school_satchel:  easy

####  Type of text #### 
:bowtie:  informative

####  Type of header #### 
:bulb:  informative

### CODE REVIEW

**Comment by Teifun2** [comment]
**Plenum:**

Problems:
- Image Artificial (2nd example)
- 1 paragraph, last sentence
- Could split into two articles

Ideas:
- CamelCase in text already

**Comment by Teifun2** [comment]
> CamelCase in text already

I would like to avoid this because i think it would really make the article annoying to read. But i can do it if required.

> Image Artificial (2nd example)

I agree that the image should not display an artificial image. 

> Could split into two articles

should i do this now or not?

**Comment by marcelluethi** [comment]
My suggestion would be to simply remove the first part. The second part can stand for itself and does not need an introduction. 

Maybe one could experiment with CamelCase in the last sentence only? Something like:

> One programming style that tries to structure this is called CamelCase. It is the practice of writing each word in the middle of a phrase with a Capital letter. 
> TheResultingTextIsMuchEasierToReadthanwhenwewouldusenocapitalizationatall.

**Comment by MoiraZuber** [comment]
Added an image suggestion. Comments?

**Comment by Teifun2** [comment]
> Added an image suggestion. Comments?

I like it very much!



> My suggestion would be to simply remove the first part. The second part can stand for itself and does not need an introduction.

Done that. I can see that it really does not need the first part.

> Maybe one could experiment with CamelCase in the last sentence only? Something like:
>
> > One programming style that tries to structure this is called CamelCase. It is the practice of writing each word in the middle of a phrase with a Capital letter.
> > TheResultingTextIsMuchEasierToReadthanwhenwewouldusenocapitalizationatall.

Because the image is no longer showing the readability improvement i added this to showcase the effect!

File: articles/article-002-camel-case.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+
+<!-- BEGIN TITLE -->
+# CamelCase, or why coding conventions matter
+<!-- END TITLE -->
+
+<!-- BEGIN BODY -->
+One problem that often arises when writing code is the naming of variables. Often multi-word names are used for variables but in most programming languages it is not possible to use spaces in a variable name. One programming style that tries to structure this is called CamelCase. It is the practice of writing each word in the middle of a phrase with a Capital letter. This greatly improves readability.
```
Comment (marcelluethi) [inline articles/article-002-camel-case.md]: Do you intentially have a the sentence "this greatly improves readability" twice (once in normal writing and once in camel case).

**Comment by marcelluethi** [comment]
I like it a lot. And the image is also really awesome.

File: articles/article-002-camel-case.md (hunk 1 above)
Comment (Teifun2) [reply articles/article-002-camel-case.md]: Nope that was an accident.

**Comment by noverney** [comment]
- add a comma after "Often multi-word names are used for variables"
- don't capitalize Capital

**Pull request merged**